crates/driver/src/run.rs

```diff
@@ -37,7 +37,11 @@ pub async fn run(
 ) {
     let args = cli::Args::parse_from(args);
     boundary::initialize_tracing(&args.log);
-    let config = config::file::load(&args.config).await;
+    let config = match (&args.config_string, &args.config_path) {
+        (Some(string), None) => config::file::load_string(string),
+        (None, Some(path)) => config::file::load_path(path).await,
+        _ => panic!("specify exactly one of config_string and config_file"),
```

Comment:
nit: I think we should use `clap` to make the options mutually exclusive if we go down this path. It should be possible and give better command line help messages when it happens.

Reply:
Yes that would be better. I didn't know it was possible. Will check clap documentation for this.

Reply:
I think the clap way of doing this is through `group` https://docs.rs/clap/latest/clap/builder/struct.Command.html#method.group but that isn't supported in the drive api yet.

Reply:
Another way of doing it would be to have one argument with a custom parsing function. Then it could be used like `--config=path=/path/to/config` and `--config=contents{foo = bar; ... }`. Not sure if that is worth the extra code.

Reply:
You should be able to use `conflicts_with`:

```rust
#[derive(Debug, Parser)]
struct Foo {
    #[clap(short, conflicts_with = "b")]
    a: bool,
    #[clap(short, conflicts_with = "a")]
    b: bool
}
```

Reply:
In theory you can use URIs and use a `data://` URI to encode it directly or `file://` URI to read from disk. Also not sure it is worth the extra code.

Reply:
`conflicts` doesn't enforce that at least one is set but I've added it.